# CIMAQ processed data analysis

In [187]:
from IPython.display import display_html 
import json
import nibabel as nib
import nilearn as nil
from nilearn.image import load_img
import os
from os import listdir as ls
from os.path import basename as bname
from os.path import dirname as dname
from os.path import join
import pandas as pd
from pandas import DataFrame as df
from cimaq_utils import loadimages


### Define main directory & visualize directory contents

### Hint:  Using a dictionary object on purpose to avoid:

#### A) Uneven lenght errors with DataFrames
#### B) Excessive/insufficient verbosity, indentation and display space with json strings

In [188]:
maindir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data"

wholedir = dict([(item, (len(ls(join(maindir, item))), ls(join(maindir, item))))
                  for item in ls(maindir)])
display(wholedir)

{'templates': (1, ['MIST_parcellation']),
 'participants': (14,
  ['sublist_3.tsv',
   'sublist_4.tsv',
   'Neuropsych',
   'sub_list.tsv',
   'TaskResults',
   'sublist_2.tsv',
   'Participants_bids.tsv',
   'MotionResults',
   'Participants_bids_headers.json',
   'Splitting_list.tsv',
   'MemoTaskParticipantFile_headers.json',
   'MemoTaskParticipantFile.tsv',
   'sublist_1.tsv',
   'sub_list_TaskQC.tsv']),
 'masks': (101,
  ['func_sub711830_mask_stereonl.nii',
   'func_sub437101_mask_stereonl.nii',
   'func_sub413474_mask_stereonl.nii',
   'func_sub164965_mask_stereonl.nii',
   'func_sub785217_mask_stereonl.nii',
   'func_sub439776_mask_stereonl.nii',
   'func_sub893978_mask_stereonl.nii',
   'func_sub630120_mask_stereonl.nii',
   'func_sub866812_mask_stereonl.nii',
   'func_sub267168_mask_stereonl.nii',
   'func_sub804743_mask_stereonl.nii',
   'func_sub127228_mask_stereonl.nii',
   'func_sub968913_mask_stereonl.nii',
   'func_sub108391_mask_stereonl.nii',
   'func_sub914042_mask_s

## Participants directory

### Participant task/trial/condition assignment & randomization

### Visualize directory contents

In [203]:
participants_dir = "../../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/participants"
filelist = sorted(ls(participants_dir))
display(filelist)

['MemoTaskParticipantFile.tsv',
 'MemoTaskParticipantFile_headers.json',
 'MotionResults',
 'Neuropsych',
 'Participants_bids.tsv',
 'Participants_bids_headers.json',
 'Splitting_list.tsv',
 'TaskResults',
 'sub_list.tsv',
 'sub_list_TaskQC.tsv',
 'sublist_1.tsv',
 'sublist_2.tsv',
 'sublist_3.tsv',
 'sublist_4.tsv']

### Preview all tsv files accessible from the 1st level in Participants directory

In [190]:
participants_lists = sorted([(file, pd.read_csv(join(participants_dir, file), sep='\t'))
                             for file in ls(participants_dir)
                             if file.endswith('.tsv')])
display(participants_lists)

[('MemoTaskParticipantFile.tsv',
       participant_id cognitive_status  clinical_age_months  schooling  sex  \
  0            108391              SCD                843.6       21.0    2   
  1            120839         Controls                940.8       14.0    2   
  2            122922              SCD                894.0       18.0    1   
  3            127228              SCD                879.6       14.0    2   
  4            139593              MCI                906.0       16.0    2   
  ..              ...              ...                  ...        ...  ...   
  98           979001         Controls                795.6       19.0    1   
  99           983291              SCD                796.8       16.0    2   
  100          988602              SCD                813.6       17.0    2   
  101          996599              MCI                820.8       13.0    2   
  102          998166         Controls                889.2       14.0    2   
  
       APOE_carr

## MemoTaskParticipantFile.tsv

## Participants_bids.tsv

## Splitting_list.tsv

## sub_list.tsv

### - All have 103 rows, thus 103 participants listed

### MemoTaskParticipantFile.tsv

In [197]:
MemoTaskParticipantFile = participants_lists[0]
# Splitting_list = participants_lists[2]
# sub_list = participants_lists[3]

display(MemoTaskParticipantFile[0], MemoTaskParticipantFile[1])

'MemoTaskParticipantFile.tsv'

participant_id cognitive_status  clinical_age_months  schooling  sex  \
0            108391              SCD                843.6       21.0    2   
1            120839         Controls                940.8       14.0    2   
2            122922              SCD                894.0       18.0    1   
3            127228              SCD                879.6       14.0    2   
4            139593              MCI                906.0       16.0    2   
..              ...              ...                  ...        ...  ...   
98           979001         Controls                795.6       19.0    1   
99           983291              SCD                796.8       16.0    2   
100          988602              SCD                813.6       17.0    2   
101          996599              MCI                820.8       13.0    2   
102          998166         Controls                889.2       14.0    2   

     APOE_carrier task_randomization QC_status  total_frames  \
0             0.0                  B       NaN           310   
1             0.0                  D       NaN           310   
2             1.0                  D       NaN           310   
3             0.0                  B       NaN           310   
4             0.0                  C       NaN           310   
..            ...                ...       ...           ...   
98            0.0                  C       NaN           310   
99            0.0                  A       NaN           310   
100           1.0                  D       NaN           310   
101           1.0                  A       NaN           310   
102           0.0                  A       NaN           288   

     total_scrubbed_frames  ...  rt_reco_Hit  rt_reco_Miss  rt_reco_New  \
0                      131  ...     1.621476      2.450333     2.582385   
1                       50  ...     2.158792      2.985500     2.069487   
2                        6  ...     2.359636      2.163913     2.191769   
3                      197  ...     2.238826      2.510000     2.228667   
4                       75  ...     2.522919      2.654415     2.499128   
..                     ...  ...          ...           ...          ...   
98                       9  ...     1.022971      1.272000     1.204538   
99                      77  ...     2.944533      3.128722     2.289667   
100                     41  ...     1.219658      1.734000     1.361000   
101                     48  ...     1.839373      3.161818     2.172077   
102                     18  ...     2.117060      2.780143     1.698205   

     rt_reco_Old  rt_reco_wrongSource  rt_reco_correctSource  rt_source_FA  \
0       1.780872             1.827200               1.525791     16.804000   
1       2.222385             2.610000               2.086016      1.103667   
2       2.301923             2.589933               2.083280      4.812400   
3       2.270115             2.302071               2.222727      2.056000   
4       2.592038             2.639591               2.351800      2.833889   
..           ...                  ...                    ...           ...   
98      1.054897             1.097100               1.010190      0.895000   
99      2.987038             5.110400               2.511360      0.471000   
100     1.232846             1.348091               1.197923      0.633250   
101     2.025872             1.978560               1.756524      1.797375   
102     2.355090             2.474333               2.004237      0.663000   

     rt_source_Hit  rt_source_wrongSource  rt_source_correctSource  
0         2.655714               4.398050                 1.845326  
1         0.862556               1.914000                 0.692968  
2         2.986782               3.402033                 2.488480  
3         2.226290               3.569929                 1.884273  
4         3.400027               3.283455                 3.571000  
..             ...                    ...                  

### Participants_bids.tsv

In [198]:
Participants_bids = participants_lists[1]
display(Participants_bids[0], Participants_bids[1])

'Participants_bids.tsv'

participant_id  clinical_age_months  schooling     sex group  \
0            108391                843.6       21.0  Female   SCD   
1            120839                940.8       14.0  Female   CTL   
2            122922                894.0       18.0    Male   SCD   
3            127228                879.6       14.0  Female   SCD   
4            139593                906.0       16.0  Female   MCI   
..              ...                  ...        ...     ...   ...   
98           979001                795.6       19.0    Male   CTL   
99           983291                796.8       16.0  Female   SCD   
100          988602                813.6       17.0  Female   SCD   
101          996599                820.8       13.0  Female   MCI   
102          998166                889.2       14.0  Female   CTL   

    group_detail site    pscid fMRI  
0            SCD  IGM  4509950  yes  
1            CTL  IGM  7424803  yes  
2           SCD+  IGM  7874568  yes  
3           SCD+  IGM  3865361  yes  
4           eMCI  JGH  5760364  yes  
..           ...  ...      ...  ...  
98           CTL  CNQ  7516889  yes  
99           SCD  CNQ  3874544  yes  
100         SCD+  IGM  6633412  yes  
101         eMCI  CNQ  4576408  yes  
102          CTL  CHS  7543023  yes  

[103 rows x 9 columns]

### Splitting_list.tsv

In [199]:
Splitting_list = participants_lists[2]
display(Splitting_list[0], Splitting_list[1])

'Splitting_list.tsv'

participant_id cognitive_status QC_status  hits  miss  false_alarms  \
0            108391              SCD       NaN    63    15             1   
1            120839         Controls       NaN    72     6             3   
2            122922              SCD       NaN    55    23             5   
3            127228              SCD       NaN    69     9             2   
4            139593              MCI       NaN    37    41             9   
..              ...              ...       ...   ...   ...           ...   
98           979001         Controls       NaN    68    10             3   
99           983291              SCD       NaN    60    18             1   
100          988602              SCD       NaN    76     2             4   
101          996599              MCI       NaN    67    11             8   
102          998166         Controls       NaN    50    28             2   

     correct_rej  old  new  correct_source  wrong_source  Fac1_memory  \
0             38   78   39              43            20      0.34777   
1             36   78   39              62            10      1.39524   
2             34   78   39              25            30      0.05356   
3             37   78   39              55            14     -0.33577   
4             30   78   39              15            22     -0.18332   
..           ...  ...  ...             ...           ...          ...   
98            36   78   39              58            10      0.83675   
99            38   78   39              50            10      0.70118   
100           35   78   39              65            11      0.05717   
101           31   78   39              42            25     -0.36577   
102           37   78   39              38            12     -0.18302   

     Fac2_semantics_and_taskSwitching  Fac3_CogInterference  Fac4_workMemo  \
0                             1.04904              -0.84308       -1.42856   
1                            -0.41944               1.55308        1.30750   
2                            -0.64087               0.10714       -1.06973   
3                             0.68150              -0.86957       -1.18414   
4                            -2.07728               0.38582        0.96525   
..                                ...                   ...            ...   
98                            0.80276               0.60318        1.06076   
99                           -0.30332              -0.70080       -0.30342   
100                           0.23326              -0.55979        0.34384   
101                          -0.71524               0.83732       -0.21984   
102                          -0.21855               0.96997       -0.00740   

    net_NaN  total_scrubbed_frames   mean_FD    dprime  associative_memScore  
0       NaN                    131  0.387678  2.818536              2.047619  
1       NaN                     50  0.273781  2.852154              4.769231  
2       NaN                      6  0.137196  1.674124              0.714286  
3       NaN                    197  0.465764  2.830928              3.437500  
4       NaN                     75  0.285148  0.671999              0.483871  
..      ...                    ...       ...       ...                   ...  
98        N                      9  0.161058  2.560993              4.461538  
99      NaN                     77  0.290982  2.685428              4.545455  
100     NaN                     41  0.189227  3.216188              4.333333  
101     NaN                     48  0.238474  1.899165              1.272727  
102       N                     18  0.177176  1.993750              2.714286  

[103 rows x 20 columns]

### sub_list.tsv

In [200]:
sub_list = participants_lists[3]
display(sub_list[0], sub_list[1])

'sub_list.tsv'

sub_ids
0     108391
1     120839
2     122922
3     127228
4     139593
..       ...
98    979001
99    983291
100   988602
101   996599
102   998166

[103 rows x 1 columns]

## However,
### sub_list_TaskQC.tsv has 94 participants

In [227]:
sub_list_TaskQC = participants_lists[4]
display(sub_list_TaskQC[0], sub_list_TaskQC[1])

'sub_list_TaskQC.tsv'

sub_ids
0    108391
1    120839
2    122922
3    127228
4    139593
..      ...
89   979001
90   983291
91   988602
92   996599
93   998166

[94 rows x 1 columns]

## sublist_1.tsv (through 4) 
### All have 24 rows (96 participants)

In [228]:
sublist_1 = participants_lists[5]
sublist_2 = participants_lists[6]
sublist_3 = participants_lists[7]
sublist_4 = participants_lists[8]

In [229]:
display_html(
             participants_lists[5][1].style.set_table_attributes(\
                "style='display:inline'").set_caption(participants_lists[5][0])._repr_html_()+
             participants_lists[6][1].style.set_table_attributes(\
                "style='display:inline'").set_caption(participants_lists[6][0])._repr_html_()+
             participants_lists[7][1].style.set_table_attributes(\
                "style='display:inline'").set_caption(participants_lists[7][0])._repr_html_()+
             participants_lists[7][1].style.set_table_attributes(\
                "style='display:inline'").set_caption(participants_lists[8][0])._repr_html_(),
             raw=True)

sublist_1.tsv sub_ids 
 
 0 
 108391 
 
 
 1 
 120839 
 
 
 2 
 122922 
 
 
 3 
 127228 
 
 
 4 
 139593 
 
 
 5 
 147863 
 
 
 6 
 150649 
 
 
 7 
 164965 
 
 
 8 
 175295 
 
 
 9 
 178101 
 
 
 10 
 189005 
 
 
 11 
 197192 
 
 
 12 
 199801 
 
 
 13 
 219637 
 
 
 14 
 229301 
 
 
 15 
 247659 
 
 
 16 
 254402 
 
 
 17 
 255499 
 
 
 18 
 258618 
 
 
 19 
 258912 
 
 
 20 
 267168 
 
 
 21 
 270218 
 
 
 22 
 271596 
 
 
 23 
 314409 
 
 sublist_2.tsv sub_ids 
 
 0 
 326073 
 
 
 1 
 336665 
 
 
 2 
 337021 
 
 
 3 
 350555 
 
 
 4 
 370092 
 
 
 5 
 385370 
 
 
 6 
 386333 
 
 
 7 
 396250 
 
 
 8 
 403131 
 
 
 9 
 408506 
 
 
 10 
 413474 
 
 
 11 
 427357 
 
 
 12 
 437101 
 
 
 13 
 439776 
 
 
 14 
 441008 
 
 
 15 
 458807 
 
 
 16 
 459801 
 
 
 17 
 462345 
 
 
 18 
 484204 
 
 
 19 
 490035 
 
 
 20 
 502616 
 
 
 21 
 517070 
 
 
 22 
 520377 
 
 
 23 
 543589 
 
 sublist_3.tsv sub_ids 
 
 0 
 549994 
 
 
 1 
 555537 
 
 
 2 
 567214 
 
 
 3 
 597569 
 
 
 4 
 619278 
 
 
 5 
 628299 
 
 
 6 
 630120 
 
 
 7 
 652850 
 
 
 8 
 658178 
 
 
 9 
 659068 
 
 
 10 
 668786 
 
 
 11 
 677561 
 
 
 12 
 711830 
 
 
 13 
 729722 
 
 
 14 
 739694 
 
 
 15 
 748676 
 
 
 16 
 763590 
 
 
 17 
 778749 
 
 
 18 
 783781 
 
 
 19 
 785217 
 
 
 20 
 785245 
 
 
 21 
 804743 
 
 
 22 
 845675 
 
 
 23 
 866812 
 
 sublist_4.tsv sub_ids 
 
 0 
 549994 
 
 
 1 
 555537 
 
 
 2 
 567214 
 
 
 3 
 597569 
 
 
 4 
 619278 
 
 
 5 
 628299 
 
 
 6 
 630120 
 
 
 7 
 652850 
 
 
 8 
 658178 
 
 
 9 
 659068 
 
 
 10 
 668786 
 
 
 11 
 677561 
 
 
 12 
 711830 
 
 
 13 
 729722 
 
 
 14 
 739694 
 
 
 15 
 748676 
 
 
 16 
 763590 
 
 
 17 
 778749 
 
 
 18 
 783781 
 
 
 19 
 785217 
 
 
 20 
 785245 
 
 
 21 
 804743 
 
 
 22 
 845675 
 
 
 23 
 866812

# Recap

In [237]:
files = [MemoTaskParticipantFile, Participants_bids, Splitting_list,
         sub_list, sub_list_TaskQC, sublist_1, sublist_2, sublist_3, sublist_4]
display(files)
# files == participants_lists

[('MemoTaskParticipantFile.tsv',
       participant_id cognitive_status  clinical_age_months  schooling  sex  \
  0            108391              SCD                843.6       21.0    2   
  1            120839         Controls                940.8       14.0    2   
  2            122922              SCD                894.0       18.0    1   
  3            127228              SCD                879.6       14.0    2   
  4            139593              MCI                906.0       16.0    2   
  ..              ...              ...                  ...        ...  ...   
  98           979001         Controls                795.6       19.0    1   
  99           983291              SCD                796.8       16.0    2   
  100          988602              SCD                813.6       17.0    2   
  101          996599              MCI                820.8       13.0    2   
  102          998166         Controls                889.2       14.0    2   
  
       APOE_carr

### Initial full list has 103 participants
### Initial partial list has 101,
### Final full list has 96 participants,
### Final partial list has 94...

## There are 7 participants who didn't meet visual QA criteria from St-Laurent (2019)
### 2 have gone missing...

## Preview all tsv files NOT accessible from the 1st level in Participants directory
#### There are 3 files

In [238]:
# Motion correction results
fMRI_meanMotion =  ("fMRI_meanMotion",
                    pd.read_csv(join(participants_dir, "MotionResults", "fMRI_meanMotion.tsv"), sep='\t'))

# Neuropsychological data
# Tables & tests are already summed up in this file
ALL_Neuropsych_scores = ("ALL_Neuropsych_scores",
                         pd.read_csv(join(participants_dir, "Neuropsych", "ALL_Neuropsych_scores.tsv"),
                                     sep='\t'))

# Memory task (fMRI task in scanner) results
fMRI_behavMemoScores =  ("fMRI_behavMemoScores", 
                pd.read_csv(join(participants_dir, "TaskResults", "fMRI_behavMemoScores.tsv"), sep='\t'))

# Visualize some examples
display(fMRI_meanMotion[0], fMRI_meanMotion[1],
        ALL_Neuropsych_scores[0], ALL_Neuropsych_scores[1],
        fMRI_behavMemoScores[0], fMRI_behavMemoScores[1])
# display(participants_jsonfiles)

'fMRI_meanMotion'

id  total_frames  total_scrubbed  mean_motion_tx  mean_motion_ty  \
0    502616         310.0             5.0        0.015959        0.040971   
1    866812         310.0           267.0        0.192033        0.139461   
2    403131         310.0           201.0        0.059968        0.114251   
3    385370         310.0            17.0        0.046048        0.044486   
4    658178         310.0            38.0        0.030602        0.038408   
..      ...           ...             ...             ...             ...   
96   878354         310.0           156.0        0.024547        0.060218   
97   785217         310.0           171.0        0.412166        1.714682   
98   437101         310.0            35.0        0.148651        0.093609   
99   258912         288.0             9.0        0.081286        0.111039   
100  517070         310.0            94.0        0.126264        0.109706   

     mean_motion_tz  mean_motion_rx  mean_motion_ry  mean_motion_rz   mean_FD  \
0          0.110429        0.090575        0.014334        0.018182  0.267941   
1          0.293642        0.405838        0.142494        0.273424  0.466325   
2          0.150056        0.172857        0.057257        0.055959  0.388421   
3          0.229207        0.099630        0.133885        0.038255  0.215581   
4          0.255382        0.247995        0.085970        0.092323  0.214392   
..              ...             ...             ...             ...       ...   
96         0.216708        0.295083        0.035390        0.039895  0.408351   
97         0.318820        2.286896        0.261100        0.548967  0.366155   
98         0.264436        0.236101        0.275109        0.173279  0.264336   
99         0.148113        0.115845        0.028842        0.089309  0.192885   
100        0.586554        0.213857        0.368406        0.117369  0.297341   

     mean_white_matt_sig  mean_ventri_sig  
0          -1.080265e-06        -0.000016  
1           1.845709e-07        -0.000013  
2          -3.124725e-06         0.000016  
3           4.346320e-06         0.000012  
4           1.062320e-06         0.000022  
..                   ...              ...  
96         -2.167879e-08        -0.000034  
97          7.891363e-08         0.000009  
98         -6.002284e-07         0.000011  
99         -1.457776e-05        -0.000017  
100         1.048202e-03         0.007009  

[101 rows x 12 columns]

'ALL_Neuropsych_scores'

dccid  hachinski_score  cdr_sb  mmse_total  moca_score  \
0    108391              1.0     0.0        25.0        30.0   
1    120839              0.0     0.0        25.0        27.0   
2    122922              0.0     0.5        25.0        28.0   
3    127228              0.0     0.0        25.0        29.0   
4    139593              0.0     0.5        26.0        23.0   
..      ...              ...     ...         ...         ...   
98   979001              0.0     0.0        26.0        29.0   
99   983291              0.0     0.0        24.0        27.0   
100  988602              0.0     0.0        26.0        27.0   
101  996599              1.0     0.5        21.0        26.0   
102  998166              1.0     0.0        23.0        25.0   

     moca_score_schooling  gds_score  WAIS_digit_symbol_total  trailA_time  \
0                    30.0       13.0                     52.0        29.96   
1                    27.0        1.0                     78.0        55.68   
2                    28.0        8.0                     66.0        38.88   
3                    29.0       12.0                     59.0        28.00   
4                    23.0       15.0                     45.0        60.84   
..                    ...        ...                      ...          ...   
98                   29.0        2.0                     73.0        29.00   
99                   27.0        4.0                     53.0        38.97   
100                  27.0        4.0                     69.0        29.72   
101                  26.0        3.0                     55.0        58.41   
102                  25.0        0.0                     54.0        38.66   

     trailB_time  ...  memoria_free_correct  memoria_total_correct  \
0          39.53  ...                   7.0                   13.0   
1         123.52  ...                   9.0                   15.0   
2          96.88  ...                   5.0                   10.0   
3          54.00  ...                   7.0                   12.0   
4         161.60  ...                   6.0                   13.0   
..           ...  ...                   ...                    ...   
98         50.87  ...                   7.0                   12.0   
99         73.25  ...                   8.0                   15.0   
100        56.76  ...                   8.0                   10.0   
101        98.00  ...                   8.0                   14.0   
102        87.97  ...                   6.0                   13.0   

     name_face_immediate_recall  name_face_delayed_recall  \
0                           7.0                       8.0   
1                           7.0                       5.0   
2                           4.0                       2.0   
3                           5.0                       5.0   
4                           2.0                       1.0   
..                          ...                       ...   
98                          3.0                       5.0   
99                          7.0                       7.0   
100                         NaN                       NaN   
101                         4.0                       5.0   
102                         5.0                       6.0   

     log_story_immediate_recall  log_story_delayed_recall  RAVLT_trial1  \
0                          16.0                      16.0           6.0   
1                          20.0                      19.0           8.0   
2                          16.0                      17.0           6.0   
3                          10.0                       6.0           6.0   
4                          12.0                      11.0           5.0   
..                          ...                       ...           ...   
98                         18.0                      19.0           7.0   
99                         14.0                      14.0           7.0   
100                        20.0                    

'fMRI_behavMemoScores'

pscID   dccID  hits  miss  false_alarms  correct_rej   old   new  \
0    7661810  490035  65.0  13.0           3.0         36.0  78.0  39.0   
1    8758136  886007  58.0  20.0           8.0         31.0  78.0  39.0   
2    7485585  386333  76.0   2.0           7.0         32.0  78.0  39.0   
3    6023087  271596  55.0  23.0           1.0         38.0  78.0  39.0   
4    8827943  845675  70.0   8.0           6.0         33.0  78.0  39.0   
..       ...     ...   ...   ...           ...          ...   ...   ...   
98   7516889  979001  68.0  10.0           3.0         36.0  78.0  39.0   
99   5089797  458807  53.0  25.0           5.0         34.0  78.0  39.0   
100  6633412  988602  76.0   2.0           4.0         35.0  78.0  39.0   
101  8278658  189005  70.0   8.0           3.0         36.0  78.0  39.0   
102  3431566  164965  53.0  25.0           2.0         37.0  78.0  39.0   

     Hits_minus_FA  Z_probabHit  ...  rt_reco_Hit  rt_reco_Miss  rt_reco_New  \
0         0.756410     0.967422  ...     1.477769      2.059231     1.801051   
1         0.538462     0.654452  ...     3.963828      6.889250     6.303769   
2         0.794872     1.949112  ...     1.587684      2.771000     2.265154   
3         0.679487     0.539208  ...     1.892818      1.700043     1.369333   
4         0.743590     1.267076  ...     1.864943      3.355500     2.012077   
..             ...          ...  ...          ...           ...          ...   
98        0.794872     1.134917  ...     1.022971      1.272000     1.204538   
99        0.551282     0.466265  ...     1.486491      2.214760     2.362103   
100       0.871795     1.949112  ...     1.219658      1.734000     1.361000   
101       0.820513     1.267076  ...     1.700714      2.160375     1.537615   
102       0.628205     0.466265  ...     2.038245      1.883360     1.296103   

     rt_reco_Old  rt_reco_wrongSource  rt_reco_correctSource  rt_source_FA  \
0       1.574679             1.323083               1.512792      1.329000   
1       4.713936             4.796308               3.723333      2.876375   
2       1.618026             1.726625               1.571338      1.582714   
3       1.835974             1.808692               1.918857      2.127000   
4       2.017821             2.091300               1.827217      1.082500   
..           ...                  ...                    ...           ...   
98      1.054897             1.097100               1.010190      0.895000   
99      1.719910             1.526556               1.465886      2.902000   
100     1.232846             1.348091               1.197923      0.633250   
101     1.747859             1.638800               1.711033      2.644333   
102     1.988603             2.493500               1.905000      2.999000   

     rt_source_Hit  rt_source_wrongSource  rt_source_correctSource  
0         1.049092               2.385583                 0.746491  
1         2.933190               5.071462                 2.315467  
2         0.897092               1.077125                 0.875912  
3         1.616436               2.422154                 1.367048  
4         0.950200               1.208400                 0.907167  
..             ...                    ...                      ...  
98        0.651074               0.844500                 0.617724  
99        1.125057               1.340278                 1.014371  
100       0.495447               0.679545                 0.464292  
101       1.269729               2.267100                 1.103500  
102       1.079453               1.780833                 0.874171  

[103 rows x 27 columns]

## ALL_Neuropsych_scores 
## fMRI_behavMemoScores
### Each have 103 rows (102 participants)
## ...

## fMRI_meanMotion
### Has only 101 (100 participants)

#### Each row being a subject, there seems to be participant attrition and/or data loss!

### Issue with the json fies
#### Files won't load

In [103]:
participants_jsonfiles = [(file, json.load(open(join(participants_dir, file))))
                          for file in ls(participants_dir)
                          if file.endswith('.json')]


JSONDecodeError: Expecting value: line 1 column 1 (char 0)

## Solution

### Open them in nano

#### A) "Participants_bids_headers.json"
#### B) "MemoTaskParticipantFile_headers.json"

## Participant metadata

In [186]:
display(participants_jsonfiles)

['Participants_bids_headers.json', 'MemoTaskParticipantFile_headers.json']

## Neuropsychological (pre-scan behavioral evaluation)

### All participants scores summed-up in a tsv file

In [ ]:

display(npsyfile)

## Motion correction analysis results

In [ ]:
display(motionfile)

In [ ]:
## Behavioral task results

In [18]:


# Niftii images diectrories
masks_dir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/masks"
allmasks = [(bname(img), load_img(img)) for img in loadimages(masks_dir)
            if "sub" in img and img.endswith(".nii")]
fmri_resample_dir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/fmri/resample"
allresampled = [(bname(img), load_img(img)) for img in loadimages(fmri_resample_dir)
            if "sub" in img and img.endswith(".nii")]
features_betamaps_dir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/features/beta_maps"

# Already sorted in https://github.com/FrancoisNadeau/cimaq_memory/blob/master/get_behav.ipynb
taskfiles_dir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/task_files/processed"

anat_dir =  "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/anat"
confouds_dir = "../../../data/cisl/DATA/cimaq_03-19/derivatives/CIMAQ_fmri_memory/data/confounds/resample"
# p1['modalities'] = [ls(join(maindir, row[0], 'ses-4'))
#               for row in p1.iterrows()]
# p1['anat'] = [ls(join(maindir, row[0], 'ses-4', 'anat'))
#               for row in p1.iterrows()]
# p1['n_anat'] = [len(row[1]['anat']) for row in p1.iterrows()]
# p1['dwi'] = [ls(join(maindir, row[0], 'ses-4', 'dwi'))
#               for row in p1.iterrows()]
# p1['func'] = [ls(join(maindir, row[0], 'ses-4', 'func'))
#               for row in p1.iterrows()]
# p1['n_dwi'] = [len(row[1]['dwi']) for row in p1.iterrows()]
# p1['n_func'] = [len(row[1]['func']) for row in p1.iterrows()]

{'templates': (1, ['MIST_parcellation']),
 'participants': (14,
  ['sublist_3.tsv',
   'sublist_4.tsv',
   'Neuropsych',
   'sub_list.tsv',
   'TaskResults',
   'sublist_2.tsv',
   'Participants_bids.tsv',
   'MotionResults',
   'Participants_bids_headers.json',
   'Splitting_list.tsv',
   'MemoTaskParticipantFile_headers.json',
   'MemoTaskParticipantFile.tsv',
   'sublist_1.tsv',
   'sub_list_TaskQC.tsv']),
 'masks': (101,
  ['func_sub711830_mask_stereonl.nii',
   'func_sub437101_mask_stereonl.nii',
   'func_sub413474_mask_stereonl.nii',
   'func_sub164965_mask_stereonl.nii',
   'func_sub785217_mask_stereonl.nii',
   'func_sub439776_mask_stereonl.nii',
   'func_sub893978_mask_stereonl.nii',
   'func_sub630120_mask_stereonl.nii',
   'func_sub866812_mask_stereonl.nii',
   'func_sub267168_mask_stereonl.nii',
   'func_sub804743_mask_stereonl.nii',
   'func_sub127228_mask_stereonl.nii',
   'func_sub968913_mask_stereonl.nii',
   'func_sub108391_mask_stereonl.nii',
   'func_sub914042_mask_s